Building WSL from source requires Developer Mode

Following the WSL developer docs [to build WSL from source]([url](https://github.com/microsoft/WSL/blob/master/doc/docs/dev-loop.md)) will fail on the creation of symbolic links unless Developer Mode is enabled in Windows:

```
Call Stack (most recent call first):
msixgluepackage/CMakeLists.txt:1 (add_appx_target)

CMake Error at cmake/FindAppx.cmake:24 (file):
file failed to create symbolic link
'D:/WSL/msixgluepackage/x64/package_layout/images/SettingsOOBEDockerDesktopIntegration.png':
A required privilege is not held by the client.
```

--- a/doc/docs/dev-loop.md
+++ b/doc/docs/dev-loop.md
@@ -16,6 +16,8 @@ The following tools are required to build WSL:
     - C++ Clang compiler for Windows
     - .NET desktop development
     - .NET WinUI app development tools
+
+- Building WSL requires support for symbolic links. To ensure this capability, enable [Developer Mode](https://learn.microsoft.com/en-us/windows/apps/get-started/enable-your-device-for-development) in Windows Settings or execute the build process with Administrator privileges.
     
 ## Building WSL
 

